### PR TITLE
SKS Line chart

### PR DESCRIPTION
**Pull request by mikolaj-jalocha**
# SKS Chart #378 

- line char has been added that shows forecasted and already measured data
- on tooltip we have 2 values along with hour of measurement
- there are some todos, I think they are not that crucial and it'd be good to close main issue and then perform some final touches. 
- the PR also solves [this](https://github.com/Solvro/mobile-topwr/issues/501) task (#501)

### CODE REVIEW

File: lib/features/sks_chart/data/repository/sks_chart_repository.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+import "../../../../api_base_rest/client/dio_client.dart";
+import "../../../../config/env.dart";
+import "../models/sks_chart_data.dart";
+
+part "sks_chart_repository.g.dart";
+
+@riverpod
+Future<IList<SksChartData>> getLatestChartData(Ref ref) async {
```
Comment (simon-the-shark) [inline lib/features/sks_chart/data/repository/sks_chart_repository.dart:12]: Did we always had the convention of `getX`? I thought we were using `XRepository` but everybody is doing `getSomething` lately

File: lib/features/sks_chart/presentation/chart_elements/sks_chart_header.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import "package:flutter/cupertino.dart";
+
+import "../../../../config/ui_config.dart";
+import "../../../../theme/app_theme.dart";
+import "../../../../utils/context_extensions.dart";
+import "../../../sks_people_live/data/models/sks_user_data.dart";
+import "../../../sks_people_live/presentation/widgets/sks_user_data_button.dart";
+
+// TODO(mikolaj-jalocha): add navigation? maybe after click on building informations
```
Comment (simon-the-shark) [inline lib/features/sks_chart/presentation/chart_elements/sks_chart_header.dart:9]: tbh questionable idea. Separate button below the menu would make more sense

File: lib/features/sks_chart/presentation/chart_elements/sks_chart_labels.dart [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+class SksChartBottomTitles extends AxisTitles {
+  SksChartBottomTitles(BuildContext context, IList<SksChartData> chartData)
+      : super(
+          sideTitles: SideTitles(
+            showTitles: true,
+            reservedSize: 35,
+            interval: 5,
+            getTitlesWidget: (double value, TitleMeta meta) {
+              final String text = (chartData.isNotEmpty)
```
Comment (simon-the-shark) [inline lib/features/sks_chart/presentation/chart_elements/sks_chart_labels.dart:44]: this line could prolly use a little refactor.

File: lib/features/sks_chart/presentation/chart_elements/sks_chart_legend.dart [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        if (isPredicted)
+          DottedBorder(
+            borderType: BorderType.RRect,
+            dashPattern: const [SksChartConfig.borderDashArray],
+            color: context.colorTheme.blueAzure,
+            padding: EdgeInsets.zero,
+            // ignore: sized_box_for_whitespace
```
Comment (simon-the-shark) [inline lib/features/sks_chart/presentation/chart_elements/sks_chart_legend.dart:46]: why did you ignore it ?

File: lib/features/sks_chart/presentation/chart_elements/sks_chart_line_touch_data.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  SksChartLineTouchData(BuildContext context, this.dateTime)
+      : super(
+          touchTooltipData: LineTouchTooltipData(
+            getTooltipColor: (LineBarSpot lineBarSpot) =>
+                context.colorTheme.greyLight,
+            getTooltipItems: (touchedSpots) {
+              return touchedSpots.map((LineBarSpot touchedSpot) {
+                final hour = (touchedSpot.barIndex == 0 ||
+                        touchedSpots.length == 1)
```
Comment (simon-the-shark) [inline lib/features/sks_chart/presentation/chart_elements/sks_chart_line_touch_data.dart:22]: some magic calculations from my perspective, could use a refactor

**Review by simon-the-shark — COMMENTED** [review]
Basically looks good to me. Good job here. Few little tips, but nothing serious

File: lib/features/sks_chart/data/repository/sks_chart_repository.dart (hunk 1 above)
Comment (mikolaj-jalocha) [reply lib/features/sks_chart/data/repository/sks_chart_repository.dart:12]: Yeah, good catch. I changed it back to follow convention

**Review by simon-the-shark — DISMISSED** [review]
:))

**Pull request merged**